backend/src/services/group/group.service.ts

```diff
@@ -152,6 +156,18 @@ class Group {
     const { id, groupsRequestDto } = data;
     const { name, permissionIds, userIds } = groupsRequestDto;
 
+    const groupById = await this.#groupsRepository.getById(id);
+
+    const isProtectedGroup =
```

Comment:
```suggestion
    const isProtectedGroup = await this.#groupsRepository.checkIsGroupProtected(id);
```

Reply:
But it's not a repository request🤔 I may create a method for a service, but I think not for a repo